tensorflow/python/keras/saving/saving_utils.py

```diff
@@ -224,7 +224,7 @@ def compile_args_from_training_config(training_config, custom_objects=None):
   # Recover loss functions and metrics.
   loss_config = training_config['loss']  # Deserialize loss class.
   if isinstance(loss_config, dict) and 'class_name' in loss_config:
-    loss_config = losses.get(loss_config)
+    loss_config = losses.get(loss_config, custom_objects)
```

Comment:
I believe the fix here should be to replace lines 226-234 with:

```python
if isinstance(loss_config, dict) and 'class_name' in loss_config:
  loss = losses.deserialize(loss_config, custom_objects)
else:
  loss = nest.map_structure(
    lambda obj: losses.deserialize(obj, custom_objects),
    loss_config)

metrics_config = training_config['metrics']
if isinstance(metrics_config, dict) and 'class_name' in metrics_config:
  metrics = metrics.deserialize(metrics_config, custom_objects)
else:
  metrics = nest.map_structure(
    lambda obj: metrics.deserialize(obj, custom_objects)
    metrics_config)

weighted_metrics_config = training_config['weighted_metrics']
if isinstance(weighted_metrics_config, dict) and 'class_name' in weighted_metrics_config:
  weighted_metrics = metrics.deserialize(weighted_metrics_config, custom_objects)
else:
  weighted_metrics = nest.map_structure(
    lambda obj: metrics.deserialize(obj, custom_objects),
    weighted_metrics_config)
```

Could you please make these changes and confirm this works for your use case?

Reply:
thanks @omalleyt12 
will do.